### ISSUE TITLE
AllenNLP has warnings about numpy incompatibilities

### ISSUE THREAD

**Issue by schmmd** [allenai/allennlp#1595]
The following warnings are output when any command is run.

```
/Users/michaels/miniconda3/envs/allennlp-pip/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)
/Users/michaels/miniconda3/envs/allennlp-pip/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192, got 176
  return f(*args, **kwds)
/Users/michaels/miniconda3/envs/allennlp-pip/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)
/Users/michaels/miniconda3/envs/allennlp-pip/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192, got 176
  return f(*args, **kwds)
```

**Comment by schmmd** [comment]
We can safely ignore this warning, but we may want to address it before the next release.

https://stackoverflow.com/questions/40845304/runtimewarning-numpy-dtype-size-changed-may-indicate-binary-incompatibility

**Comment by matt-gardner** [comment]
These warnings come from the user's environment, and are fixable by fixing your environment.  I don't think we should address this ourselves.

**Comment by schmmd** [comment]
@matt-gardner I think we need to address the problem in the dependencies we install.  I get these warnings in a fresh environment following our instructions, and we also have this same problem in our Docker environment.

```
$ docker run -it allennlp/commit:b79d500108f61f42319910afb08cf9e98ea7dea5
/usr/local/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)
/usr/local/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192, got 176
  return f(*args, **kwds)
/usr/local/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)
/usr/local/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192, got 176
  return f(*args, **kwds)
```

**Comment by schmmd** [comment]
@matt-gardner were you able to fix this in your environment?

Maybe we can resolve this by upgrading some of our dependencies?

**Comment by matt-gardner** [comment]
Well, I thought I had fixed this is my environment when I wrote that, then I discovered that it wasn't actually fixed. So, apologies, you can ignore my last comment.

**Comment by schmmd** [comment]
These warnings go away if you downgrade numpy, `pip install numpy==1.14.3`.

**Comment by joelgrus** [comment]
fixed by #1601
